modules/brushingandlinking/src/brushingandlinkingmanager.cpp

```diff
@@ -394,12 +402,18 @@ int BrushingAndLinkingManager::getActionIndex(BrushingAction action) {
 void BrushingAndLinkingManager::propagate(BrushingAction action, BrushingTarget target) {
     modifications_[target] |= fromAction(action);
 
-    if (std::holds_alternative<BrushingAndLinkingOutport*>(owner_)) {
-        auto outport = std::get<BrushingAndLinkingOutport*>(owner_);
-        outport->getProcessor()->invalidate(invalidationLevel_);
-    } else if (!parent_) {
-        auto inport = std::get<BrushingAndLinkingInport*>(owner_);
-        inport->getProcessor()->invalidate(invalidationLevel_);
+    // Only invalidate the top level in the connected brushing manager network
+    if (!parent_) {
+        if (std::holds_alternative<BrushingAndLinkingOutport*>(owner_)) {
+            auto outport = std::get<BrushingAndLinkingOutport*>(owner_);
+            // Processor need to be invalidated for the network evaluation to be notified.
+            outport->getProcessor()->invalidate(invalidationLevel_);
```

Comment:
Any reason for nog doing? 
```suggestion
            outport->invalidate(invalidationLevel_);
```

Reply:
See comment the line above:
Processor need to be invalidated for the network evaluation to be notified.

Reply:
Then you need too specify why that processor needs to invalidate and not the one connected to the outport

Reply:
I just added a comment to make it a bit clearer than before :) Agree that we should clarify even more though
I think that @martinfalk needed `notifyObserversInvalidationBegin(this);` to be called for some use case.

Reply:
The network should get evaluated either way so which call you make has nothing to do with the network being evaluated. The only question is if the processor owning the outport needs to also be invalidated for some reason, or if it is enough that the down stream processors are evaluated.

Reply:
Yes, I think that was also one of the desired behaviours.